### ISSUE TITLE
Make in-place parameter changes take effect, or fail loudly (Model._run_from_state takes self as static)

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#735]
Noticed while working on #732 / #733. Filing separately so it does not get buried in a provenance PR.

This is documented behaviour, not a regression: `Model._run_from_state` is decorated `@partial(jax.jit, static_argnums=(0, 4, 5, 6, 8, 9))`, and the decorator already carries the note `# Note: changing fields assumed static won't propagate.` The enhancement is to make the consequence either impossible or loud, because today it is silent.

## What the current behaviour costs

Because `self` is static, every physics term and its parameters are baked into the compiled executable as constants. Changing a parameter in place afterwards does not change the model's identity for cache purposes, so JAX reuses the existing executable and the new value never reaches the computation.

Measured at T31L8 SPEEDY, half a day, using `trvdi` (the vertical-diffusion timescale), with a control to show the parameter is not simply inert:

```python
m = Model(coords=coords, physics=speedy_physics(), time_step=30.0)
T1 = m.run(save_interval=0.5, total_time=0.5).dynamics.temperature

term = next(t for t in m.physics.terms if t.name == "speedy_vertical_diffusion")
term.params.set_value(term.params.get_value().replace(trvdi=jnp.array(2.0)))

T2 = m.run(save_interval=0.5, total_time=0.5).dynamics.temperature
```

```
CONTROL  fresh Model built with trvdi=2 : max|dT| vs baseline = 1.570 K
IN-PLACE mutation to trvdi=2           : max|dT| vs run 1     = 0.0
```

So the parameter moves the answer by 1.57 K when the model is rebuilt, and by nothing at all when it is mutated in place. Rebuilding the `Model` is the correct workaround today.

## Why it is worth addressing

This is the natural way to write a sensitivity or calibration loop:

```python
for value in candidates:
    set_parameter(model, value)
    loss = evaluate(model.run(...))
```

Every iteration returns the first iteration's physics. The loop runs, costs full compute, and produces a flat response surface with no error and no warning, which reads as "this parameter does not matter" rather than "this loop did nothing".

## Current mitigation

#733 makes it visible but does not fix it: parameters are captured at trace time, so the record reports the values that actually ran, and a divergence between the live module and the compiled ones raises a warning plus a `live_parameters_differ_from_compiled` key in the record.

## Possible directions

- Pass the parameter state into the jit dynamically (`nnx.split` / `nnx.state` as a traced argument, merged inside) so parameters are traced values rather than baked constants. This is the real fix and would also make gradients with respect to parameters natural, but it touches the compiled hot path and needs a performance check first.
- Cheaper: raise or warn at the API level when `run` is called on a model whose parameter state differs from the one its cached executable was traced with. #733 already computes exactly this comparison, so it would mostly be a matter of moving where it fires.

Either way it is worth a line in `CLAUDE.md` alongside the other JAX gotchas, since the working approach (rebuild the `Model` to change a parameter) is not currently discoverable.
